sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -474,4 +474,30 @@ export class InstitutionService extends RecordDataModelService<Institution> {
       await queryRunner.release();
     }
   }
+
+  /**
+   * Search the institution based on the search criteria.
+   * @param legalName legalName of the institution.
+   * @param operatingName operatingName of the institution.
+   * @returns Searched institution details.
+   */
+  async searchInstitution(
+    legalName: string,
+    operatingName: string,
+  ): Promise<Institution[]> {
+    const searchQuery = this.repo
+      .createQueryBuilder("institution")
+      .where("institution.establishedDate is not null");
```

Comment:
Is this default condition added to handle the scenario where none of the search criteria is entered? If so IMO I would recommend API throwing error where !legalName && !operatingName

Reply:
If search criteria is not given, the search cannot be triggered from the UI, the search button is disabled.

Reply:
But what if someone plays with API? to secure the API.

Reply:
:) Yes they can until they have AEST authorization and query params by design are always optional, else we would have used path params. If both the values are not given, all the rows will be returned.

Reply:
IMO. Still API must validate the same what UI has done(value to at least one of the search params). @andrewsignori-aot  @ann-aot please share your opinion here.

Reply:
I am not following the discussion. The API has a check to validate it, the same as the UI. What am I missing here?
![image](https://user-images.githubusercontent.com/61259237/142286318-dd50dbba-a1d3-44da-a128-d53096fc0d82.png)

Reply:
My bad. I did not check the API 😒
Sorry.

Reply:
All good @dheepak-aot, these are exactly the questions that we need keep asking 😉